tests/test_collection.py

```diff
@@ -109,6 +109,27 @@ def test_Collection_sortUpdate(collection):
         collection.sortUpdate(sort="bad-sort")
     collection.sortUpdate("release")
 
+@pytest.mark.authenticated
+def test_Collection_visibility(collection):
+    visibility = collection.visibility()
+    with pytest.raises(BadRequest):
+        visibility.move()
+    with pytest.raises(BadRequest):
+        visibility.remove()
+    visibility.updateVisibility(recommended=True, home=True, shared=True)
+    assert visibility.promotedToRecommended is True
+    assert visibility.promotedToOwnHome is True
+    assert visibility.promotedToSharedHome is True
+    visibility.updateVisibility(recommended=False, home=False, shared=False)
+    assert visibility.promotedToRecommended is False
+    assert visibility.promotedToOwnHome is False
+    assert visibility.promotedToSharedHome is False
+    visibility.move()
+    visibility.remove()
+    with pytest.raises(BadRequest):
+        visibility.move()
+    with pytest.raises(BadRequest):
```

Comment:
Actually raises `NotFound`.
```suggestion
    with pytest.raises(NotFound):
```